### ISSUE TITLE
False positive report of `UnusedMethod` for methods used with JUnit 5 `MethodSource` annotation

### ISSUE THREAD

**Issue by proggeo** [google/error-prone#2335]
### Description of the problem / feature request:

Error prone flags static method as unused if it's used only in JUnit 5 MethodSource annotation ([docs](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource))

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
class MyTest {
    private enum MyEnum {
        ALPHA,
        BETA
    }

    @ParameterizedTest
    @MethodSource("provideParameters")
    void testModification(MyEnum enumValue, boolean booleanValue) {
        System.out.println(enumValue);
        System.out.println(booleanValue);
    }

    private static Stream<Arguments> provideParameters() {
        return Arrays.stream(MyEnum.values())
                .flatMap(val -> StreamEx.of(Arguments.of(val, true), Arguments.of(val, false)));
    }
}
```

`warning: [UnusedMethod] Private method 'provideParameters' is never used.`

### What version of Error Prone are you using?

2.6.0

###  Have you found anything relevant by searching the web?

Previously reported within this (now closed) issue - https://github.com/google/error-prone/issues/1500#issuecomment-622382947

**Comment by proggeo** [comment]
thanks @cushon
